File: surf-chat/src/main/java/dev/slne/surf/social/chat/command/TogglePmCommand.java
package dev.slne.surf.social.chat.command;

import dev.jorel.commandapi.CommandAPICommand;
import dev.slne.surf.social.chat.SurfChat;
import dev.slne.surf.social.chat.object.ChatUser;
import dev.slne.surf.social.chat.util.MessageBuilder;

public class TogglePmCommand extends CommandAPICommand {

  public TogglePmCommand(String commandName) {
    super(commandName);

    withPermission("surf.chat.command.toggle");
    executesPlayer((player, args) -> {
      ChatUser user = ChatUser.getUser(player.getUniqueId());

      if(user.isToggledPM()) {
        user.setToggledPM(false);
        SurfChat.send(player, new MessageBuilder().primary("Du hast privat Nachrichten ").success("aktiviert."));
      } else {
        user.setToggledPM(true);
        SurfChat.send(player, new MessageBuilder().primary("Du hast privat Nachrichten ").error("deaktiviert."));
      }
    });
  }
}

File: surf-chat/src/main/java/dev/slne/surf/social/chat/command/TogglePmCommand.kt
package dev.slne.surf.social.chat.command

import com.github.shynixn.mccoroutine.bukkit.launch
import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.executors.CommandArguments
import dev.jorel.commandapi.executors.PlayerCommandExecutor
import dev.jorel.commandapi.kotlindsl.playerExecutor
import dev.slne.surf.social.chat.SurfChat
import dev.slne.surf.social.chat.`object`.ChatUser
import dev.slne.surf.social.chat.util.MessageBuilder
import org.bukkit.entity.Player

class TogglePmCommand(commandName: String) : CommandAPICommand(commandName) {
    init {
        withPermission("surf.chat.command.toggle")

        playerExecutor{ player, _ ->
            SurfChat.instance.launch {
                val user: ChatUser = ChatUser.getUser(player.uniqueId)

                if (user.toggledPM) {
                    user.toggledPM = false
                    SurfChat.send(player, MessageBuilder().primary("Du hast privat Nachrichten ").success("aktiviert."))
                } else {
                    user.toggledPM = true
                    SurfChat.send(player, MessageBuilder().primary("Du hast privat Nachrichten ").error("deaktiviert."))
                }
            }
        }
    }
}
